### ISSUE TITLE
[v2] Field Array + touched / resetForm

### ISSUE THREAD

**Issue by OfficerHalf** [jaredpalmer/formik#2022]
## 🐛 Bug report

tldr; Probably related to #1712 but the behavior is slightly different since I'm setting touched manually. Shape of FormikTouched should differ if values are arrays - `boolean[] | undefined` instead of `boolean | undefined`. If I ignore the type errors, it works as expected.

### Current Behavior

As part of submission I am calling resetForm with values and touched. My form is like this (simplified as much as possible):

```
// mail, givenName, and sn are all always arrays with a single value
// so I only ever look at the first element. Thanks, LDAP!
interface User {
  mail: string[];
  givenName: string[];
  c: string;
  sn: string[];
}

const emptyUser = () => {
  return { mail: [""], givenName: [""], c: "", sn: [""] };
};

// ...

return <Formik initialValues={{ users: [emptyUser()] }}>
  {formikProps => (
    <Form>
      <FieldArray>
        {helpers => (
          <>
            {formikProps.values.users.map((user, index) => {
              return (
                <div>
                  <Field name={`users[${index}].mail[0]`}>
                    {fieldProps => /* render field */}
                  </Field>
                  <Field name={`users[${index}].givenName[0]`} >
                    {fieldProps => /* render field */}
                  </Field>
                  <Field name={`users[${index}].sn[0]`} >
                    {fieldProps => /* render field */}
                  </Field>
                  <Field name={`users[${index}].c`} >
                    {fieldProps => /* render field */}
                  </Field>
                </div>
              );
            })}
            <button
              type="button"
              onClick={helpers.handlePush(emptyUser())}
            >
              Add User
            </button>
            <button type="button" onClick={formikProps.handleSubmit}>
              Submit
            </button>
          </>
        )}
      </FieldArray>
    </Form>
  )}
</Formik>
```

The touched value I get from each Field's props is correct - it's a single value for the first element in the array for mail, givenName, sn. I can inspect the whole form's touched value and it matches what I expect, something like:

```
{
  users: [{
    mail: [true],
    givenName: [true],
    sn: [true],
    c: true
  }]
}
```

However, when calling resetForm in handleSubmit, it expects single values instead of arrays:

```
const handleSubmit = (values: { users: User[] }, formikHelpers: FormikHelpers<{ users: User[] }>) => {
  formikHelpers.setSubmitting(true);
  const invalidUsers: User[] = [];
  const invalidUsersTouched: FormikTouched<User>[] = [];
  const newUsersPromise = values.users.map(user => {
    if (UserSchema.isValidSync(user)) {
      // make an async call with the user
      return submitUser(user); // this is a promise
    } else {
      invalidUsers.push(user);
      // ** THE ISSUE **
      // Formik expects single values for these instead of arrays for mail, givenName, and sn
      invalidUsersTouched.push({ mail: true, givenName: true, c: true, sn: true });
    }
  });
  Promise.all(newUsersPromise).then(() => {
    formikHelpers.setSubmitting(false);

    if (invalidUsers.length > 0) {
      formikHelpers.resetForm({ values: { users: invalidUsers }, touched: { users: invalidUsersTouched } });
    } else {
      formikHelpers.resetForm();
    }
  });
};
```

And inspecting touched after submission shows:

```
{
  users: [
    {
      c: true,
      givenName: true,
      mail: true,
      sn: true
    }
  ]
}
```

Each field's touched value is also incorrect. If I inspect `fieldProps.meta.touched`, it's false. If I ignore the type error, everything works as expected:

```
// Typescript complains that boolean[] is not assignable to boolean | undefined.
invalidUsersTouched.push({ mail: [true], givenName: [true], c: true, sn: [true] });
```

### Expected behavior

Typing for touched should respect array types.

### Reproducible example

Sandbox wasn't playing nice with Formik v2 and Typescript, sorry.

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Formik           | 2.0.3
| React            | 16.8.6
| TypeScript       | 3.5.1? i think...
| Browser          | Edge Dev (Chromium)
| npm/Yarn         | 6.12.0
| Operating System | Win 10

**Comment by theryanclausen** [comment]
Has this been addressed?
